packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -321,16 +326,25 @@ class AttachCommand extends FlutterCommand {
         result = await app.runner.waitForAppToFinish();
         assert(result != null);
       } else {
-        final Completer<void> onAppStart = Completer<void>.sync();
-        unawaited(onAppStart.future.whenComplete(() {
-          TerminalHandler(runner)
-            ..setupTerminal()
-            ..registerSignalHandlers();
-        }));
-        result = await runner.attach(
-          appStartedCompleter: onAppStart,
-        );
-        assert(result != null);
+        while (true) {
```

Comment:
What happens on the second iteration of this loop when `ProtocolDiscovery` is not used above?

Reply:
it now breaks if the stream is closed.